### ISSUE TITLE
Xcode 9.1 deprecation warnings

### ISSUE THREAD

**Issue by OlivierLapraye** [Hearst-DD/ObjectMapper#913]
I just updated Xcode to the newest 9.1 version, and have 4 deprecation warnings (I use ObjectMapper v3.0.0)  :
```
(3) 'characters' is deprecated: Please use String or Substring directly.
(1) 'CharacterView' is deprecated: Please use String or Substring directly.
```

As you probably know already, Swift 4 works without `characters`, directly with the String. For example, `hex.characters.count` can be replaced by `hex.count`.

Here are the concerned line of code :
- `characters` appears in `HexColorTransform.swift` line **38** and **92**.
- `characters` appears in `ToJson.swift` line **37**
- `CharacterView` appears in `ToJson.swift` line **44**

**Comment by joebenton** [comment]
+1 :)

**Comment by lieonCX** [comment]
+1:)

**Comment by tarzon** [comment]
+1 :)

**Comment by tristanhimmelman** [comment]
I've just released a version with the 9.1 fixes in it
